src/common/pageUtils.ts

```diff
@@ -59,3 +59,14 @@ export async function sendRequest(
 
     return response;
 }
+
+export async function getIDFromURL(
+    element: Puppeteer.ElementHandle,
+    selector: string
+): Promise<number> {
+    const url = await element.$eval(selector, (anchor: Element) =>
+        anchor.getAttribute("href")
+    );
+    const urlParts: string[] = ("" + url).split("/");
```

Comment:
I'm not sure what's `""+url` for?

Reply:
if it's to cast to string you can do so with something more TS friendly like: `string(url)`

Reply:
instead i added a null check